pajbot/modules/vip_refresh.py

```diff
@@ -37,53 +32,37 @@ def update_vip_cmd(self, bot, source, **rest):
         # TODO if you wanted to improve this: Provide the user with feedback
         #   whether the update succeeded, and if yes, how many users were updated
         bot.whisper(source, "Reloading list of VIPs...")
-        self._update_vips()
+        bot.action_queue.submit(self._update_vips)
 
     def _update_vips(self):
-        self.bot.privmsg("/vips")
-
-    @staticmethod
-    def _filter_vips(vips: List[str]) -> List[str]:
-        return [vip for vip in vips if re.match(VIPRefreshModule.LOGIN_REGEX, vip)]
-
-    @staticmethod
-    def _parse_pubnotice_for_vips(msg_id, message):
-        if msg_id == "no_vips":
-            return []
-
-        if msg_id == "vips_success":
-            if message.startswith("The VIPs of this channel are: ") and message.endswith("."):
-                message = message[30:-1]  # 30 = length of the above "prefix", -1 removes the dot at the end
-                vips = message.split(", ")
-                # Response of "vips_success" returns display names, which could contain non-ascii characters.
-                # For now, we filter these out since these cannot be resolved through helix, see https://github.com/pajbot/pajbot/issues/1772
-                filtered_vips = VIPRefreshModule._filter_vips(vips)
-                if len(vips) != len(filtered_vips):
-                    log.warning("Some non-ascii display names were filtered out from vips_success NOTICE response.")
-                return filtered_vips
-            log.warning(
-                f"Received vips_success NOTICE message, but actual message did not begin with expected prefix. Message was: {message}"
-            )
-
-        return None
-
-    def _on_pubnotice(self, channel: str, msg_id, message) -> None:
         if self.bot is None:
-            log.warn("_on_pubnotice failed in VIPRefreshModule because bot is None")
-            return
-
-        if channel != self.bot.streamer.login:
-            return
+            log.error("_update_vips failed in VIPRefreshModule because bot is None")
+        return
 
-        vip_logins = self._parse_pubnotice_for_vips(msg_id, message)
+        try:
+            vip_ids = self.bot.twitch_helix_api.fetch_all_vips(
+                self.bot.streamer.id, self.bot.streamer_access_token_manager
+            )
 
-        if vip_logins is not None:
-            self.bot.action_queue.submit(self._process_vip_logins, vip_logins)
+            if vip_ids is not None:
```

Comment:
`vip_ids` can never be `None`, ideally this would be reflected in the typing of the method as well